pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp

```diff
@@ -1326,6 +1329,76 @@ class ConvertAtenLinearOp : public ConvertAtenMatmulBaseOp<AtenOpT> {
   }
 };
 
+template <>
+LogicalResult ConvertAtenOp<AtenSqueezeDimOp>::matchAndRewrite(
+    AtenSqueezeDimOp op,
+    OpAdaptor adaptor,
+    ConversionPatternRewriter& rewriter) const {
+  auto self = adaptor.self();
+  auto selfTy = self.getType().template cast<RankedTensorType>();
+  if (!selfTy)
+    return op.emitError("Only ranked tensor types supported in MHLO");
+  int64_t dim;
+  if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+    return rewriter.notifyMatchFailure(
+        op, "Only constant dim is currently supported");
+
+  auto rank = selfTy.getRank();
+  if (rank == 0) {
+    return rewriter.notifyMatchFailure(
+        op, "The rank of tensor must be greater than 0");
+  }
+
+  dim = (dim + rank) % rank;
```

Comment:
`torch.squeeze` also support `dim=None` in which case all size-1 dims are removed. But it seems that we only sequeeze on explicity size-1 dim ?

Reply:
The `AtenSqueezeDimOp` here only means squeeze the specified dimension.